replica/table.cc

```diff
@@ -733,6 +735,31 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
     co_return co_await with_scheduling_group(_config.memtable_scheduling_group, std::ref(try_flush));
 }
 
+future<> table::maybe_wait_for_sstable_count_reduction() {
+    while (true) {
+        if (_async_gate.is_closed()) {
+            co_return;
+        }
+        const auto sstable_runs_with_memtable_origin = boost::copy_range<std::unordered_set<utils::UUID>>(
+            *_main_sstables->all()
+            | boost::adaptors::filtered([] (const sstables::shared_sstable& sst) {
+                return sst->get_origin() == "memtable";
+            })
+            | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::run_identifier)));
+        const auto threshold = std::max(schema()->max_compaction_threshold(), 32);
+        if (sstable_runs_with_memtable_origin.size() <= threshold || is_auto_compaction_disabled_by_user()) {
+            co_return;
+        }
+        trigger_compaction();
+        auto start = db_clock::now();
+        co_await _sstables_changed.wait();
```

Comment:
This is a condition variable, yes? Those are susceptible to spurious wakeup. They should always be enclosed in a loop. Our condition variable supports looping via an [overload](http://docs.seastar.io/master/classseastar_1_1condition__variable.html#a91ffa147b82c0d1419a2643696d34038).